dask/dataframe/core.py

```diff
@@ -2193,18 +2213,64 @@ def std(
             )
             return handle_out(out, result)
         else:
-            v = self.var(skipna=skipna, ddof=ddof, split_every=split_every)
+            v = numeric_dd.var(skipna=skipna, ddof=ddof, split_every=split_every)
             name = self._token_prefix + "std"
+
+            if needs_time_conversion:
+                sqrt_func_kwargs = {
+                    "is_df_like": is_df_like,
+                    "time_cols": time_cols if is_df_like else None,
+                    "axis": axis,
+                }
+                sqrt_func = _sqrt_and_convert_to_timedelta
+            else:
+                sqrt_func_kwargs = {}
+                sqrt_func = np.sqrt
+
             result = map_partitions(
-                np.sqrt,
+                sqrt_func,
                 v,
                 meta=meta,
                 token=name,
                 enforce_metadata=False,
                 parent_meta=self._meta,
+                **sqrt_func_kwargs,
             )
+
+            if is_df_like:
+                result = result.astype(meta.dtype)
```

Comment:
This feels a little strange to me. Is meta guaranteed to have a `dtype`?

Reply:
Not sure :/. Since this function is `@derived_from(pd.DataFrame)`, is the dtype from it not included? `meta` is created from `_meta_nonempty._std()` which I think outputs a Series? I am honestly still not sure how to get the resulting dtypes to match other than to do this.

One thing I can do is check if `meta` has the `dtype` attribute at least before doing this.

Reply:
This has been included in the new commit!